core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/api/percentile.kt

```diff
@@ -0,0 +1,294 @@
+package org.jetbrains.kotlinx.dataframe.api
+
+import org.jetbrains.kotlinx.dataframe.AnyRow
+import org.jetbrains.kotlinx.dataframe.ColumnsSelector
+import org.jetbrains.kotlinx.dataframe.DataColumn
+import org.jetbrains.kotlinx.dataframe.DataFrame
+import org.jetbrains.kotlinx.dataframe.DataRow
+import org.jetbrains.kotlinx.dataframe.RowExpression
+import org.jetbrains.kotlinx.dataframe.aggregation.ColumnsForAggregateSelector
+import org.jetbrains.kotlinx.dataframe.annotations.AccessApiOverload
+import org.jetbrains.kotlinx.dataframe.columns.ColumnReference
+import org.jetbrains.kotlinx.dataframe.columns.toColumnSet
+import org.jetbrains.kotlinx.dataframe.impl.aggregation.aggregators.Aggregators
+import org.jetbrains.kotlinx.dataframe.impl.aggregation.aggregators.cast
+import org.jetbrains.kotlinx.dataframe.impl.aggregation.interComparableColumns
+import org.jetbrains.kotlinx.dataframe.impl.aggregation.modes.aggregateAll
+import org.jetbrains.kotlinx.dataframe.impl.aggregation.modes.aggregateFor
+import org.jetbrains.kotlinx.dataframe.impl.aggregation.modes.aggregateOf
+import org.jetbrains.kotlinx.dataframe.impl.aggregation.modes.of
+import org.jetbrains.kotlinx.dataframe.impl.columns.toComparableColumns
+import org.jetbrains.kotlinx.dataframe.impl.suggestIfNull
+import org.jetbrains.kotlinx.dataframe.math.percentileOrNull
+import kotlin.reflect.KProperty
+
+// region DataColumn
+
+public fun <T : Comparable<T>> DataColumn<T?>.percentile(percentile: Double): T =
+    percentileOrNull(percentile).suggestIfNull("percentile")
+
+public fun <T : Comparable<T>> DataColumn<T?>.percentileOrNull(percentile: Double): T? =
+    Aggregators.percentile(percentile).cast<T>().aggregate(this)
+
+public inline fun <T, reified R : Comparable<R>> DataColumn<T>.percentileOfOrNull(
+    percentile: Double,
+    noinline expression: (T) -> R?,
+): R? = Aggregators.percentile(percentile).cast<R?>().aggregateOf(this, expression)
+
+public inline fun <T, reified R : Comparable<R>> DataColumn<T>.percentileOf(
+    percentile: Double,
+    noinline expression: (T) -> R?,
+): R = percentileOfOrNull(percentile, expression).suggestIfNull("percentileOf")
+
+// endregion
+
+// region DataRow
+
+public fun AnyRow.rowPercentileOrNull(percentile: Double): Any? =
+    Aggregators.percentile(percentile).aggregate(
+        values().filterIsInstance<Comparable<Any?>>().toValueColumn(),
```

Comment:
This will break when you have a `String` and `Number` column for instance. While they both are `Comparable`, they are not comparable to each other. I see we make the same mistake in `rowMin` etc. (it's [noted](https://github.com/Kotlin/dataframe/issues/961#issuecomment-2659202472)). It's probably best to filter for `Number` values, like `rowMean`.

Reply:
Let's add an unified solution for all these functions

Reply:
sure! I'd probably go with a route like this, similar to `describeImpl` with `convertToComparableOrNull`:
- First filter for all `Comparable<*>` values in the row
- If all these values are comparable to each other, run the statistic function on it
- If not, filter for `Number` values. These are not directly comparable to each other but can be made comparable by converting them to Double/BigDecimal, and thus all statistics functions will work. Though, then we skip any other potentially comparable values, like `String`.
- Else return `null`

I think this would produce the most expected behavior. wdyt?

Reply:
Looks nice! But let's do it in a new PR for all statistical functions.

Reply:
sure! Then I'd make this `Number` for now until we fix it for all